superset/common/query_object.py

```diff
@@ -94,6 +96,7 @@ class QueryObject:
     datasource: Optional[BaseDatasource]
     result_type: Optional[ChartDataResultType]
     is_rowcount: bool
+    time_offset: List[str]
```

Comment:
I think it would be great to deprecate the current string-based offset, and use a periods-based approach, much like in rolling window, using the time grain as the time offset unit.